backend/amd64/reg_class.ml

```diff
@@ -67,8 +90,8 @@ module T = struct
     let reg_class = of_machtype typ in
     let index = reg_index - first_available_register reg_class in
     match reg_class with
-    | Int64 -> index
-    | Float128 ->
+    | GPR -> index
+    | SIMD ->
       let slot_size_in_vals = 2 in
```

Comment:
This looks suspicious, or deserves
a comment since `SIMD` is no longer
guaranteed to be 128 bits.)

Reply:
This is wrong, good catch. It needs to know about the live set now, since the offset depends on whether vec256/512s are in use, so I made it specific to amd64 reg_class.

Reply:
(Also note this value is only used for computing the offset of `Valx2` registers with autovectorization)